### PR TITLE
Mimecast audit

### PR DESCRIPTION
**Pull request by nipun-crestdatasystem**
This is a draft PR to run the validation checks for MimecastAudit.

### CODE REVIEW

**Comment by v-prasadboke** [comment]
Hello @nipun-crestdatasystem, Thank you for raising this PR. Given that this PR is still in draft state, could you kindly let us know when it will be ready for review?

**Comment by v-prasadboke** [comment]
Hello @nipun-crestdatasystem, As this PR is moved to Ready for review state so I will examine this PR and update you about the same before 31 August, 2023. Thanks.

**Comment by v-prasadboke** [comment]
Hello @nipun-crestdatasystem, Thank you for raising this PR. This PR will be investigated and will update you about the same before 01 September, 2023.

File: Solutions/MimecastAudit/Data/Solution_MimecastAudit.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    "Name": "MimecastAudit",
+    "Author": "Mimecast - dlapi@mimecast.com",
+    "Logo": "<img src=\"https://raw.githubusercontent.com/mimecast/Azure-Sentinel/mimecast_audit_connector/Solutions/MimecastAudit/Data%20Connectors/Logo/Mimecast.svg\" width=\"75px\" height=\"75px\">",
```
Comment (v-prasadboke) [inline Solutions/MimecastAudit/Data/Solution_MimecastAudit.json:4]: The logo link should have the master branch, which is currently mimecast_audit_connector.
Please switch it to master.

File: Solutions/MimecastAudit/Data/Solution_MimecastAudit.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+    "Name": "MimecastAudit",
+    "Author": "Mimecast - dlapi@mimecast.com",
+    "Logo": "<img src=\"https://raw.githubusercontent.com/mimecast/Azure-Sentinel/mimecast_audit_connector/Solutions/MimecastAudit/Data%20Connectors/Logo/Mimecast.svg\" width=\"75px\" height=\"75px\">",
+    "Description": "The data connector for Mimecast Audit & Authentication provides customers with the visibility into security events related to audit and authentication events within Microsoft Sentinel. The data connector provides pre-created dashboards to allow analysts to view insight into user activity, aid in incident correlation and reduce investigation response times coupled with custom alert capabilities.  \nThe Mimecast products included within the connector are: \nAudit & Authentication\n\n\nMicrosoft Sentinel Solutions provide a consolidated way to acquire Microsoft Sentinel content like data connectors, workbooks, analytics, and automations in your workspace with a single deployment step.",
```
Comment (v-prasadboke) [inline Solutions/MimecastAudit/Data/Solution_MimecastAudit.json:5]: Period ( . ) is expected after Audit & Authentication.

File: Solutions/MimecastAudit/Data Connectors/MimecastAudit_API_AzureFunctionApp.json [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
...tAccessKey: Access Key for the dedicated Mimecast user\n - mimecastSecretKey: Secret Key for dedicated Mimecast user\n - mimecastBaseURL: Regional Mimecast API Base URL\n - activeDirectoryAppId: Azure portal ---> App registrations ---> [your_app] ---> Application ID\n - activeDirectoryAppSecret: Azure portal ---> App registrations ---> [your_app] ---> Certificates & secrets ---> [your_app_secret]\n\n >Note: If using Azure Key Vault secrets for any of the values above, use the`@Microsoft.KeyVault(SecretUri={Security Identifier})`schema in place of the string values. Refer to [Key Vault references documentation](https://docs.microsoft.com/azure/app-service/app-service-key-vault-references) for further details.\n\n4. Mark the checkbox labeled **I agree to the terms and conditions stated above**. \n5. Click **Purchase** to deploy. \n\n6. Go to ***Azure portal ---> Resource groups ---> [your_resource_group] --->  [appName](type: Storage account) ---> Storage Explorer ---> BLOB CONTAINERS ---> Audit checkpoints ---> Upload*** and create empty file on your machine named checkpoint.txt and select it for upload (this is done so that date_range for SIEM logs is stored in consistent state)\n"
```
Comment (v-prasadboke) [inline Solutions/MimecastAudit/Data Connectors/MimecastAudit_API_AzureFunctionApp.json:118]: Here ( appInsightsLocation(default): westeurope ) in data connector description it is said the default location is westeurope. 
Do the parameter appInsightsLocation should have Default location value as West Europe?

File: Solutions/MimecastAudit/SolutionMetadata.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+    "publisherId": "azuresentinel",
```
Comment (v-prasadboke) [inline Solutions/MimecastAudit/SolutionMetadata.json:2]: Does your partner center supports this solution id, if not change the solution id.

**Comment by v-prasadboke** [comment]
Also please workbook metadata in https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V2/WorkbookMetadata/WorkbooksMetadata.json .
Thanks.

**Comment by v-prasadboke** [comment]
After making all the necessary changes, please repackage the solution using V3 tool.

**Comment by v-prasadboke** [comment]
Hello @nipun-crestdatasystem, Thank you for committing the requested changes. We will Examine this changes and update you about the same before 07 September, 2023.

File: Solutions/MimecastAudit/Data Connectors/azuredeploy_MimecastAudit_AzureFunctionApp.json [hunk 5]
```diff
@@ -0,0 +1,466 @@
...
+            "active_directory_app_id": "[concat('@Microsoft.KeyVault(SecretUri=https://', variables('functionAppName'), '.vault.azure.net/secrets/', 'active-directory-app-id', '/)')]",
+            "active_directory_app_secret": "[concat('@Microsoft.KeyVault(SecretUri=https://', variables('functionAppName'), '.vault.azure.net/secrets/', 'active-directory-app-secret', '/)')]",
+            "active_directory_tenant_id": "[concat('@Microsoft.KeyVault(SecretUri=https://', variables('functionAppName'), '.vault.azure.net/secrets/', 'active-directory-tenant-id', '/)')]",
+            "log_analytics_workspace_id": "[concat('@Microsoft.KeyVault(SecretUri=https://', variables('functionAppName'), '.vault.azure.net/secrets/', 'log-analytics-workspace-id', '/)')]",
+            "log_analytics_workspace_key": "[concat('@Microsoft.KeyVault(SecretUri=https://', variables('functionAppName'), '.vault.azure.net/secrets/', 'log-analytics-workspace-key', '/)')]",
+            "WEBSITE_RUN_FROM_PACKAGE": "https://github.com/mimecast/Azure-Sentinel/blob/mimecast_audit_connector/Solutions/MimecastAudit/Data%20Connectors/MimecastAuditAzureConn.zip"
```
Comment (v-prasadboke) [inline Solutions/MimecastAudit/Data Connectors/azuredeploy_MimecastAudit_AzureFunctionApp.json:209]: Please change the branch to master

**Comment by v-prasadboke** [comment]
Hello @nipun-crestdatasystem, Please commit the above requested change. Other than all looks good, Also please resolve the branch conflicts

**Review by v-prasadboke — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Comment by nipun-crestdatasystem** [comment]
@v-atulyadav Once the PR was merged we created the offer and published it for the preview audience, but when the user clicks on the Azure Deploy button he is navigated to the Bing home page. Seems like the template link and the zip link are not updated can you please add the respective links?
image (13)

**Pull request merged**